plugins/workload-management/src/main/java/org/opensearch/plugin/wlm/service/QueryGroupPersistenceService.java

```diff
@@ -0,0 +1,186 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.plugin.wlm.service;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.ClusterStateUpdateTask;
+import org.opensearch.cluster.metadata.Metadata;
+import org.opensearch.cluster.metadata.QueryGroup;
+import org.opensearch.cluster.service.ClusterManagerTaskThrottler.ThrottlingKey;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.Priority;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.core.action.ActionListener;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.plugin.wlm.action.CreateQueryGroupResponse;
+import org.opensearch.search.ResourceType;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+/**
+ * This class defines the functions for QueryGroup persistence
+ */
+public class QueryGroupPersistenceService {
+    private static final String SOURCE = "query-group-persistence-service";
+    private static final String CREATE_QUERY_GROUP_THROTTLING_KEY = "create-query-group";
+    private static final Logger logger = LogManager.getLogger(QueryGroupPersistenceService.class);
+    /**
+     *  max QueryGroup count setting name
+     */
+    public static final String QUERY_GROUP_COUNT_SETTING_NAME = "node.query_group.max_count";
+    /**
+     * default max queryGroup count on any node at any given point in time
+     */
+    public static final int DEFAULT_MAX_QUERY_GROUP_COUNT_VALUE = 100;
+    /**
+     *  max QueryGroup count setting
+     */
+    public static final Setting<Integer> MAX_QUERY_GROUP_COUNT = Setting.intSetting(
+        QUERY_GROUP_COUNT_SETTING_NAME,
+        DEFAULT_MAX_QUERY_GROUP_COUNT_VALUE,
+        0,
+        (newVal) -> {
+            if (newVal > 100 || newVal < 1) throw new IllegalArgumentException(
```

Comment:
Validation values should be constant. Also, this logic should be delegated to a validation method here and in the `setMaxQueryGroupCount` method to avoid redundancy